src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -501,8 +499,7 @@ boolean canRepair(
       try {
         Map<String, String> dcByNode = segment.getReplicas();
 
-        return !isRepairRunningOnNodes(segment, dcByNode, keyspace, cluster)
-            && nodesReadyForNewRepair(coordinator, segment, dcByNode, busyHosts);
```

Comment:
the method signature can be simplified now to `boolean canRepair(RepairSegment segment, JmxProxy coordinator) {`

Reply:
It'll even become simpler with just `canRepair(RepairSegment segment)`: https://github.com/thelastpickle/cassandra-reaper/pull/996/files#diff-2e6b3c333eecade8a9d85f64416757e3edd4bbf4f1b67f9ebd60d7dd780f899aR457